Assets/MixedRealityToolkit-SDK/Features/SpatialAwareness/System/MixedRealitySpatialAwarenessSystem.cs

```diff
@@ -0,0 +1,572 @@
+            };
+        
+        #endregion Surface Finding Events
+
+        #endregion IMixedRealityManager Implementation
+
+        #region IMixedRealtyEventSystem Implementation
+
+        /// <inheritdoc />
+        public override void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.EventFunction<T> eventHandler)
+        {
+            base.HandleEvent(eventData, eventHandler);
+        }
+
+        /// <summary>
+        /// Registers the <see cref="GameObject"/> to listen for boundary events.
+        /// </summary>
+        /// <param name="listener"></param>
+        public override void Register(GameObject listener)
+        {
+            base.Register(listener);
+        }
+
+        /// <summary>
+        /// UnRegisters the <see cref="GameObject"/> to listen for boundary events.
+        /// /// </summary>
+        /// <param name="listener"></param>
+        public override void Unregister(GameObject listener)
+        {
+            base.Unregister(listener);
+        }
+
+        #endregion
+
+        #region IMixedRealityEventSource Implementation
+
+        /// <inheritdoc />
+        bool IEqualityComparer.Equals(object x, object y)
+        {
+            // There shouldn't be other Boundary Managers to compare to.
+            return false;
+        }
+
+        /// <inheritdoc />
+        public int GetHashCode(object obj)
+        {
+            return Mathf.Abs(SourceName.GetHashCode());
+        }
+
+        /// <inheritdoc />
+        public uint SourceId { get; } = 0;
+
+        /// <inheritdoc />
+        public string SourceName { get; } = "Mixed Reality Spatial Awareness System";
+
+        #endregion IMixedRealityEventSource Implementation
+
+        #region IMixedRealitySpatialAwarenessSystem Implementation
+
+        /// <inheritdoc />
+        public bool StartObserverSuspended { get; set; } = false;
+
+        /// <inheritdoc />
+        public Vector3 ObservationExtents { get; set; } = Vector3.one * 10;
+
+        private float updateInterval = 3.5f;
+
+        /// <inheritdoc />
+        public float UpdateInterval
+        {
+            get
+            {
+                return updateInterval;
+            }
+
+            set
+            {
+                if (IsObserverRunning)
+                {
+                    throw new System.InvalidOperationException("UpdateInterval cannot be modified while the observer is running.");
+                }
+
+                updateInterval = value;
+            }
+        }
+
+        /// <inheritdoc />
+        public bool IsObserverRunning { get; private set; } = false;
+
+        /// <inheritdoc />
+        public void ResumeObserver()
+        {
+            if (IsObserverRunning) { return; }
+
+#if UNITY_WSA
+            // todo
+#endif // UNITY_WSA
+
+            IsObserverRunning = true;
+        }
+
+        /// <inheritdoc />
+        public void SuspendObserver()
+        {
+            if (!IsObserverRunning) { return; }
+
+#if UNITY_WSA
```

Comment:
We shouldn't have any platform specific code in our main system, but in platform components.

Reply:
Similar to how we did devices for input system.

Reply:
we likely will have to have something to trigger the correct observer. this is purely placeholder for now